### DISCUSSION TITLE
Updates to seqr's search infrastructure

### DISCUSSION THREAD

**Discussion by lynnpais** [broadinstitute/seqr#3890] [Feature Updates]
We have migrated _seqr_’s underlying infrastructure from Elasticsearch to Hail to prioritize variant-level information in our data representation. This also allows for a more scalable and versatile data model to support the growing _seqr_ community.

#### New Features:

Global Allele Count: Users can now view a global count of alleles from all projects in _seqr_, providing a clearer perspective on both common and rare genetic variants. When looking up a variant, you will only see details about the case if you have access to it.

Variant Lookup Tool: The [Variant Lookup](https://seqr.broadinstitute.org/summary_data/variant_lookup) tool will enable users to explore any variant present in _seqr_ using genomic coordinates. In the near future, features will be added to view and share high-level information such as inheritance, zygosity, and phenotype.

Automatic Reference Updates: Data stays more up-to-date with the latest information from ClinVar and other reference databases through automatic data updates.

#### Ongoing Enhancements:
While we are continually striving to improve _seqr_, we acknowledge that certain searches, such as a recessive search may experience slower performance. One adaptation we have found helpful is to have the recessive search run in a new window while you are reviewing the results from the dominant search. If you encounter a 504 error with a search, try re-running it. If the search is no longer completing successfully, please submit [a ticket](https://github.com/broadinstitute/seqr/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=) including the specific error message. Your patience and feedback are highly valued.

#### Note for Local Instances:
For users running their own instance of _seqr_ locally, rest assured that we will continue to support Elasticsearch for the foreseeable future. Later in the year, we will provide comprehensive documentation to guide you through migrating your local _seqr_ instance to the new Hail backend. Ample time will be provided to facilitate a smooth transition before the Elasticsearch-based installations stop receiving updates.

**Comment by ghost** [comment]
Are these features available as public commits in https://github.com/broadinstitute/seqr, https://github.com/broadinstitute/seqr-helm, or https://github.com/broadinstitute/seqr-loading-pipelines?  What about docs? 😄
Thanks for all the hard dev work on this feature!

Cheers,
Rob

**Reply by hanars** [reply]
Hi Rob,

Per the "Note for Local Instances" in our announcement
> Later in the year, we will provide comprehensive documentation to guide you through migrating your local seqr instance to the new Hail backend

For now, we recommend that you do not try to migrate to the new code on your own, as the underlying data model is still being iterated on and it is highly likely that if you migrate now you will need to migrate again. That said, all the code is available in the seqr master branch

Best,
Hana

**Reply by ghost** [reply]
Thanks Hana.  I should have reread the post before commenting.

Cheers 🍻

**Comment by ghost** [comment]
Hi @hanars and @lynnpais.  I hope you are doing well!

Are there any updates regarding the hail search feature and docs?

**Reply by hanars** [reply]
Hi Rob, we will post in the announcements section once there is anything to announce! As of now, no updates
